webapp/src/components/tutorial_tour_tip/tutorial_tour_tip.tsx

```diff
@@ -0,0 +1,330 @@
+import 'tippy.js/animations/scale-subtle.css'
+import 'tippy.js/animations/perspective-subtle.css'
+
+import {Placement} from 'tippy.js'
+
+import CloseIcon from '../../widgets/icons/close'
+import Button from '../../widgets/buttons/button'
+import IconButton from '../../widgets/buttons/iconButton'
+import CompassIcon from '../../widgets/icons/compassIcon'
+
+import PulsatingDot from '../pulsating_dot'
+
+import TutorialTourTipBackdrop, {Coords, TutorialTourTipPunchout} from './tutorial_tour_tip_backdrop'
+
+import useTutorialTourTipManager, {TutorialTourTipManagerProps} from './tutorial_tour_tip_manager'
+
+const TourTipOverlay = ({
+    children,
+    show,
+    onClick,
+}: { children: React.ReactNode; show: boolean; onClick: (e: React.MouseEvent) => void }) =>
+    (show ? ReactDOM.createPortal(
+        <div
+            className='tutorial-tour-tip__overlay'
+            onClick={onClick}
+        >
+            {children}
+        </div>,
+        document.body,
+    ) : null)
+
+type Props = {
+    screen: JSX.Element
+    title: JSX.Element
+    imageURL?: string
+    punchOut?: TutorialTourTipPunchout | null
+    step: number
+    singleTip?: boolean
+    showOptOut?: boolean
+    placement?: Placement
+    telemetryTag?: string
+    stopPropagation?: boolean
+    preventDefault?: boolean
+    tutorialCategory: string
+    onNextNavigateTo?: () => void
+    onPrevNavigateTo?: () => void
+    autoTour?: boolean
+    pulsatingDotPosition?: Coords | undefined
+    width?: string | number
+    className?: string
+    hideNavButtons?: boolean
+    hideBackdrop?: boolean
+    clickThroughPunchhole?: boolean
+    onPunchholeClick?: (e: React.MouseEvent) => void
+    skipCategoryFromBackdrop?: boolean
+}
+
+const TutorialTourTip = ({
+    title,
+    screen,
+    imageURL,
+    punchOut,
+    autoTour,
+    tutorialCategory,
+    singleTip,
+    step,
+    onNextNavigateTo,
+    onPrevNavigateTo,
+    telemetryTag,
+    placement,
+    showOptOut,
+    pulsatingDotPosition,
+    stopPropagation = true,
+    preventDefault = true,
+    width = 320,
+    className,
+    hideNavButtons = false,
+    hideBackdrop = false,
+    onPunchholeClick,
+    skipCategoryFromBackdrop,
+}: Props): JSX.Element => {
+    const managerProps: TutorialTourTipManagerProps = {
+        step,
+        autoTour,
+        telemetryTag,
+        tutorialCategory,
+        onNextNavigateTo,
+        onPrevNavigateTo,
+        stopPropagation,
+        preventDefault,
+    }
+
+    const triggerRef = useRef(null)
+    const {
+        show,
+        tourSteps,
+        handleOpen,
+        handleHide,
+        handleDismiss,
+        handleNext,
+        handlePrevious,
+        handleSkipTutorial,
+        handleSavePreferences,
+        getLastStep,
+        handleEventPropagationAndDefault,
+        handleSendToNextTour,
+    } = useTutorialTourTipManager(managerProps)
+
+    const getButtonText = (): JSX.Element => {
+        let buttonText = (
+            <>
```

Comment:
Why this have a fragment for the if only has one element?

Reply:
Done.